.github/workflows/ci.yml

```diff
@@ -62,30 +89,88 @@ jobs:
         # The .ts scripts depend upon this
       - run: npm install --global tsx
 
-      - name: build test images
+      - name: build test images ${{ matrix.platform }} == ${{ env.DEFAULT_PLATFORM }}
         run: |
           docker info
-          ./buildTestImages.ts
+          ./buildTestImages.ts ${{ matrix.platform == env.DEFAULT_PLATFORM && ' ' || '--dry' }}
       - name: run test images
+        if: ${{ matrix.platform == env.DEFAULT_PLATFORM }}
         run: ./runTestImages.ts
 
-      # XXX this should be instant because all the stages were already built in the steps above
-      # but it's re-building the last stage. This is deemed good enough for now.
-      # see https://github.com/moby/moby/issues/34715
-      - name: Build and push complete image
+      - name: Compute tags
+        id: docker-tags
+        run: |
+          sep=
+          SUFFIXED=
+          uarch=$(echo "${{ matrix.platform }}" | tr / _)
+          for TAG in ${{ steps.meta.outputs.tags }}; do
+            SUFFIXED="$SUFFIXED$sep$TAG-$uarch"
+            if test -z "$sep"; then
+              sep=,
+              echo "tag=$TAG-$uarch" >> $GITHUB_OUTPUT
+            fi
+          done
+          echo "tags=$SUFFIXED" >> $GITHUB_OUTPUT
+      
+      # XXX this should be instant for the local platform because all the stages
+      # were already built in the steps above but it's re-building the last
+      # stage. This is deemed good enough for now.  see
+      # https://github.com/moby/moby/issues/34715
+      - name: Build and push images
         uses: docker/build-push-action@v5
         with:
           context: .
+          platforms: ${{ matrix.platform }}
           # push to registry on every repo push. A PR #2 will push with tag `pr-2` and `main` will have tag `main`.
           # See https://github.com/docker/metadata-action?tab=readme-ov-file#basic.
           push: true
-          tags: ${{ steps.meta.outputs.tags }}
+          tags: ${{ steps.docker-tags.outputs.tags }}
           labels: ${{ steps.meta.outputs.labels }}
-      - name: notify on failure
-        if: failure() && github.event_name != 'pull_request'
-        uses: ./.github/actions/notify-status
```

Comment:
This seems to be completely gone, any reason to remove master failure notifications ?